src/libs/SidebarUtils.js

```diff
@@ -0,0 +1,315 @@
+        if (ReportUtils.isDefaultRoom(report) && !Permissions.canUseDefaultRooms(betas) && ReportUtils.getPolicyType(report, policies) !== CONST.POLICY.TYPE.FREE) {
+            return false;
+        }
+
+        if (ReportUtils.isUserCreatedPolicyRoom(report) && !Permissions.canUsePolicyRooms(betas)) {
+            return false;
+        }
+
+        if (isPolicyExpenseChat && !Permissions.canUsePolicyExpenseChat(betas)) {
+            return false;
+        }
+
+        return true;
+    });
+
+    let orderedReports = _.sortBy(filteredReports, sortByTimestampDescending ? 'lastMessageTimestamp' : 'reportName');
+
+    if (sortByTimestampDescending) {
+        orderedReports.reverse();
+    }
+
+    // Move the archived Rooms to the last
+    orderedReports = _.sortBy(orderedReports, report => ReportUtils.isArchivedRoom(report));
+
+    // Put all the reports into the different buckets
+    for (let i = 0; i < orderedReports.length; i++) {
+        const report = orderedReports[i];
+
+        // If the report is pinned and we are using the option to display pinned reports on top then we need to
+        // collect the pinned reports so we can sort them alphabetically once they are collected
+        if (report.isPinned) {
+            pinnedReportOptions.push(report);
+        } else if (report.hasOutstandingIOU && !report.isIOUReportOwner) {
+            iouDebtReportOptions.push(report);
+
+        // If the active report has a draft, we do not put it in the group of draft reports because we want it to maintain it's current position. Otherwise the report's position
+        // jumps around in the LHN and it's kind of confusing to the user to see the LHN reorder when they start typing a comment on a report.
+        } else if (report.hasDraft && report.reportID.toString() !== currentlyViewedReportID) {
+            draftReportOptions.push(report);
+        } else {
+            recentReportOptions.push(report);
+        }
+    }
+
+    // Prioritizing reports with draft comments, add them before the normal recent report options
+    // and sort them by report name.
+    const sortedDraftReports = _.sortBy(draftReportOptions, 'text');
+    recentReportOptions = sortedDraftReports.concat(recentReportOptions);
+
+    // Prioritizing IOUs the user owes, add them before the normal recent report options and reports
+    // with draft comments.
+    const sortedIOUReports = _.sortBy(iouDebtReportOptions, 'iouReportAmount').reverse();
+    recentReportOptions = sortedIOUReports.concat(recentReportOptions);
+
+    // If we are prioritizing our pinned reports then shift them to the front and sort them by report name
+    const sortedPinnedReports = _.sortBy(pinnedReportOptions, 'text');
+    recentReportOptions = sortedPinnedReports.concat(recentReportOptions);
+
+    return _.chain(recentReportOptions)
+        .pluck('reportID')
+        .map(reportID => reportID.toString())
```

Comment:
NAB, I feel like we could theoretically avoid this` _.map()` if we used some different data structure like a `Map`. But unsure if it would be worth looking into to reduce an iteration.

Reply:
Yeah, we could just push the `reportID` into the arrays instead of the entire report objects. Maybe I'll do that in the followup issue

Reply:
I made a note of this in https://github.com/Expensify/Expensify/issues/229617